infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/AsyncStorageModeValidator.java

```diff
@@ -0,0 +1,158 @@
+    this.defaultNamespaceName = defaultNamespaceName;
+    this.runtimesPerUser = runtimesPerUser;
+  }
+
+  @Override
+  public void validate(Map<String, String> attributes) throws ValidationException {
+    if (parseBoolean(attributes.get(ASYNC_PERSIST_ATTRIBUTE))) {
+      isEphemeralAttributeValidation(attributes);
+      pvcStrategyValidation();
+      alowUserDefinedNamespaceValidation();
+      nameSpaceStrategyValidation();
+      runtimesPerUserValidation();
+    }
+  }
+
+  @Override
+  public void validateUpdate(Map<String, String> existing, Map<String, String> update)
+      throws ValidationException {
+    if (parseBoolean(update.get(ASYNC_PERSIST_ATTRIBUTE))) {
+      if (isEphemeral(existing) || isEphemeral(update)) {
+        pvcStrategyValidation();
+        alowUserDefinedNamespaceValidation();
+        nameSpaceStrategyValidation();
+        runtimesPerUserValidation();
+      } else {
+        String message =
+            "Workspace configuration not valid: Asynchronous storage available only for NOT persistent storage";
+        LOG.warn(message);
+        throw new ValidationException(message);
+      }
+    }
+  }
+
+  private void isEphemeralAttributeValidation(Map<String, String> attributes)
+      throws ValidationException {
+    if (!isEphemeral(attributes)) {
+      String message =
+          "Workspace configuration not valid: Asynchronous storage available only for NOT persistent storage";
+      LOG.warn(message);
+      throw new ValidationException(message);
+    }
+  }
+
+  private void runtimesPerUserValidation() throws ValidationException {
+    if (runtimesPerUser > 1) {
+      String message =
+          format(
+              "Workspace configuration not valid: Asynchronous storage available only if 'che.limits.user.workspaces.run.count' set to 1, but got %s",
+              runtimesPerUser);
+      LOG.warn(message);
+      throw new ValidationException(message);
+    }
+  }
+
+  private void nameSpaceStrategyValidation() throws ValidationException {
+    if (Strings.isNullOrEmpty(defaultNamespaceName)
+        || !defaultNamespaceName.contains("<username>")) {
+      String message =
+          "Workspace configuration not valid: Asynchronous storage available only for 'per-user' namespace strategy";
+      LOG.warn(message);
+      throw new ValidationException(message);
+    }
+  }
+
+  private void alowUserDefinedNamespaceValidation() throws ValidationException {
+    if (allowUserDefinedNamespaces) {
+      String message =
+          format(
+              "Workspace configuration not valid: Asynchronous storage available only if 'che.infra.kubernetes.namespace.allow_user_defined' set to 'false', but got '%s'",
+              allowUserDefinedNamespaces);
+      LOG.warn(message);
+      throw new ValidationException(message);
+    }
+  }
+
+  private void pvcStrategyValidation() throws ValidationException {
+    if (!COMMON_STRATEGY.equals(pvcStrategy)) {
```

Comment:
This can be evaluated in the constructor just once.

Reply:
https://github.com/eclipse/che/issues/17420